File: saxpy.go
package main

func govecSaxpy(N int, alpha float32, X []float32, incX int, Y []float32, incY int) (int) {

	var xi, yi, i int

	for i = 0; i < N; i++ {
		Y[yi] += alpha * X[xi]
		xi += incX
		yi += incY
	}

	return 0
}

File: test/govec/saxpy.c
void  govecSaxpy(int N, float alpha, float * X, int incX, float * Y, int incY) {

	int xi, yi, i;

	xi = 0;
	yi = 0;

	for( i = 0; i < N; i++ ){
		Y[yi] += alpha * X[xi];
		xi += incX;
		yi += incY;
	};
}
